src/cache/fs.ts

```diff
@@ -12,8 +12,12 @@ export default class FSCache extends BaseCacheAdapter {
     }
   }
 
-  resolve (key: string) {
-    return resolve(this.options.cache.dir, key)
+  key (src: string) {
+    return shortHash(src.split('#').shift()?.split('//').pop())
```

Comment:
- URL parsing logic belongs to remote adaptor
- We can use ufo.parseURL to ensure there are no duplicate cache entries with different encoding (hash removing is also strange since browser should not pass it at all) 
- IPX is not currently hashing so cache entries are readable (we could use ufo.encodeSegment to ensure fs issue is not happening) but in case of hashing, i think it would be much better doing it in src/ipx.ts (so new adaptors use same hashing) (well ideally we have to unuse class syntax and have generateHash(src) util :D)